src/main/java/org/testng/internal/annotations/IAnnotationFinder.java

```diff
@@ -46,6 +47,15 @@
    */
   <A extends IAnnotation> A findAnnotation(Constructor<?> cons, Class<A> annotationClass);
 
+  /**
+   *
+   * @param cls - The corresponding class.
+   * @param annotationClass - The class on which annotation is to be looked for.
+   * @param <A> - The expected {@link IAnnotation} type
+   * @return The annotation on the method. If not found, return the annotation on the declaring
+   *     class. If not found, return null.
```

Comment:
If not found, should return an empty list (it is what is expected and what the code is doing)

Reply:
done